internal/s3/client.go

```diff
@@ -60,32 +67,65 @@ func (c *Client) Put(bucket, key string, payload goja.ArrayBuffer) PutResponse {
 }
 
 func (c *Client) Get(bucket, key string) GetResponse {
-	var (
-		buf = make([]byte, 4*1024)
-		sz  int
-	)
 	stats.Report(c.vu, objGetTotal, 1)
 	start := time.Now()
-	obj, err := c.cli.GetObject(context.Background(), &s3.GetObjectInput{
-		Bucket: aws.String(bucket),
-		Key:    aws.String(key),
+
+	var objSize = 0
+	err := get(c.cli, bucket, key, func(chunk []byte) {
+		objSize += len(chunk)
 	})
 	if err != nil {
 		stats.Report(c.vu, objGetFails, 1)
 		return GetResponse{Success: false, Error: err.Error()}
 	}
+
 	stats.Report(c.vu, objGetDuration, metrics.D(time.Since(start)))
+	stats.ReportDataReceived(c.vu, float64(objSize))
+	return GetResponse{Success: true}
+}
+
+func get(
+	c *s3.Client,
+	bucket string,
+	key string,
+	onDataChunk func(chunk []byte),
+) error {
+	var buf = make([]byte, 4*1024)
+
+	obj, err := c.GetObject(context.Background(), &s3.GetObjectInput{
+		Bucket: aws.String(bucket),
+		Key:    aws.String(key),
+	})
+	if err != nil {
+		return err
+	}
+
 	for {
 		n, err := obj.Body.Read(buf)
 		if n > 0 {
-			sz += n
+			onDataChunk(buf[:n])
 		}
 		if err != nil {
 			break
 		}
 	}
-	stats.ReportDataReceived(c.vu, float64(sz))
-	return GetResponse{Success: true}
+	return nil
+}
+
+func (c *Client) VerifyHash(bucket, key, expectedHash string) VerifyHashResponse {
```

Comment:
Maybe this method still needs to report some metrics, because why not? It receives data so it can count size. It may be quite useful for the report. The same applicable for native protocol client.